tracee-ebpf/tracee/tracee.go

```diff
@@ -2086,3 +2104,228 @@ func readArgFromBuff(dataBuff io.Reader) (argTag, interface{}, error) {
 	}
 	return argTag, res, nil
 }
+
+func InsertSortUniq(d []string, s string) []string {
+	p := sort.SearchStrings(d, s)
+	if p < len(d) { // exists
+		return d
+	}
+	d = append(d, "")
+	copy(d[p+1:], d[p:]) // copy to search index
+	d[p] = s
+	return d
+}
+
+func DeleteSortUniq(d []string, s string) []string {
+	p := sort.SearchStrings(d, s)
+	if p == len(d) { // does not exist
+		return d
+	}
+	d = append(d[:p], d[:p+1]...)
+	return d
+}
+
+// Get information about running containers
+
+type Containers struct {
+	cgroupv1   bool
+	cgroupv1mp string
+	cgroupv2   bool
+	cgroupv2mp string
+	mapContIds map[string][]string
+}
+
+// manage containers and pids within those containers
+
+func (c *Containers) _AddContainer(contid string) {
+	// add container
+	if c.mapContIds == nil {
+		c.mapContIds = make(map[string][]string, 1)
+	}
+	_, ok := c.mapContIds[contid]
+	if !ok {
+		c.mapContIds[contid] = []string{}
+	}
+}
+
+func (c *Containers) _DelContainer(contid string) {
+	// remove container
+	_, ok := c.mapContIds[contid]
+	if ok {
+		delete(c.mapContIds, contid)
+	}
+}
+
+func (c *Containers) _CheckContainer(contid string) bool {
+	// check if container exists
+	_, ok := c.mapContIds[contid]
+	if !ok {
+		return false
+	}
+	return true
+}
+
+func (c *Containers) GetContainers() []string {
+	// list existing containers
+	var conts []string
+	for k, _ := range c.mapContIds {
+		conts = append(conts, k)
+	}
+	if len(conts) == 0 {
+		return nil
+	}
+	return conts
+}
+
+func (c *Containers) _AddPid(contid string, pid string) {
+	// add pid to container
+	c._AddContainer(contid)
+	c.mapContIds[contid] = InsertSortUniq(c.mapContIds[contid], pid)
+}
+
+func (c *Containers) _DelPid(contid string, pid string) {
+	// del pid from container
+	if c._CheckContainer(contid) == true {
+		c.mapContIds[contid] = DeleteSortUniq(c.mapContIds[contid], pid)
+	}
+}
+
+func (c *Containers) GetPids(contid string) []string {
+	// list pids from existing container
+	if c._CheckContainer(contid) == false {
+		return nil
+	}
+	return c.mapContIds[contid]
+}
+
+// Populate info
+
+func (c *Containers) Populate() error {
+	var err error
+
+	err = c._ProcMountsCgroups()
+	if err != nil {
+		return err
+	}
+	err = c._PopulateDocker()
+	if err != nil {
+		return err
+	}
+	err = c._PopulatePodman()
+	if err != nil {
+		return err
+	}
+
```

Comment:
We should add a third populate that doesn't contain any prefix (e.g. docker/libpod) in its regex (I think this should work for cri-o/containerd)